packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -207,11 +423,97 @@ class FlutterError extends AssertionError {
   ///
   /// All sentences in the error should be correctly punctuated (i.e.,
   /// do end the error message with a period).
-  @override
-  String get message => super.message;
+  ///
+  /// This constructor defers to the [new FlutterError.fromParts] constructor.
+  /// The first line is wrapped in an implied [ErrorSummary], and subsequent
+  /// lines are wrapped in implied [ErrorDescription]s. Consider using the
+  /// [new FlutterError.fromParts] constructor to provide more detail, e.g.
+  /// using [ErrorHint]s or other [DiagnosticsNode]s.
+  factory FlutterError(String message) {
+    final List<String> lines = message.split('\n');
+    final List<DiagnosticsNode> parts = <DiagnosticsNode>[];
+    parts.add(ErrorSummary(lines.first));
+    if (lines.length > 1)  {
+      parts.addAll(lines.skip(1).map<DiagnosticsNode>((String line) => ErrorDescription(line)));
+    }
+    return FlutterError.fromParts(parts);
+  }
 
+  /// Create an error message from a list of [DiagnosticsNode]s.
+  ///
+  /// By convention, there should be exactly one [FlutterSummary] in the list,
+  /// and it should be the first entry.
+  ///
+  /// Other entries are typically [ErrorDescription]s (for material that is
+  /// always applicable for this error) and [ErrorHint]s (for material that may
+  /// be sometimes useful, but may not always apply). Other [DiagnosticsNode]
+  /// subclasses, such as [DiagnosticsStackTrace], may
+  /// also be used.
+  FlutterError.fromParts(this.diagnostics) : assert(diagnostics.isNotEmpty, FlutterError.fromParts(<DiagnosticsNode>[ErrorSummary('Empty FlutterError')])) {
```

Comment:
oh there is something beautiful about the FlutterError constructor's assert using itself.

Reply:
I believe in dogfooding all the features I write. :)